File: June 2021/June_17.java
class Solution {
    public int numSubarrayBoundedMax(int[] nums, int left, int right) {
        int start = -1, end = -1, ans = 0;
        for(int i = 0; i < nums.length; i++){
            if(nums[i] > right){
                start = i;
                end = i;
            } else if(nums[i] >= left && nums[i] <= right)
                end = i;
            ans += end - start;
        }
        return ans;
    }
}

File: June 2022/June_17.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int ans;
    
    int helper(TreeNode* root){
        int left = (root->left == nullptr ? 1 : helper(root->left));
        int right = (root->right == nullptr ? 1 : helper(root->right));
        if(left == 2 || right == 2){
            ans++;
            return 0;
        }
        if(left == 0 || right == 0)
            return 1;
        return 2;
    }
    
    int minCameraCover(TreeNode* root) {
        ans = 0;
        if(helper(root) == 2)
            ans++;
        return ans;
    }
};
